File: LP/c/5.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_FRASE 100

int main(){

    char frase[MAX_FRASE],compare[MAX_FRASE], *words, *palavras[MAX_FRASE];
    int flag = 0,counter = 0;

    printf("insira uma frase: \n");
    fgets(frase, MAX_FRASE, stdin);
    printf("insira uma palavra: \n");
    fgets(compare, MAX_FRASE, stdin);

    words = strtok(frase, " ");

    for(int i = 0; words != NULL; i++) { 
        palavras[i] = words;

        for(int c = 0; c < strlen(palavras[i]); c++){

            for(int t = 0; t < strlen(compare); t++){

                if(palavras[i][c+t] == compare[t]){
                    flag++;

                    if(flag == strlen(compare)-1){
                        counter++;
                        flag = 0;
                    } 

                }else{flag = 0;}
            }
        }
        words = strtok(NULL, " ");
    }
    printf("Existem %d ocorrencias da palavra na frase.\n", counter);
    return 0;
}

File: LP/go/5.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "strings"
)

func main() {
    scanner := bufio.NewScanner(os.Stdin)
    counter := 0
    fmt.Println("Digite uma frase:")
    scanner.Scan()
    frase := scanner.Text()

    fmt.Println("Digite uma palavra para procurar:")
    scanner.Scan()
    palavra := scanner.Text()

    palavras := strings.Split(frase, " ")
    for i := 0; i< len(palavras); i++{
        counter += contarOcorrencias(palavras[i], palavra)
    }
    fmt.Printf("A palavra '%s' ocorre %d vezes na frase.\n", palavra, counter)
}

func contarOcorrencias(frase, palavra string) int {
   
    ocorrencias := 0
    index := strings.Index(frase, palavra)

    for index != -1 {
        ocorrencias++
        index = strings.Index(frase[index+1:], palavra)
        if index != -1 {
            index += ocorrencias * len(palavra)
        }
    }
    return ocorrencias
}